includes/User.php

```diff
@@ -3173,34 +3132,29 @@ public function doLogout() {
 	 * @todo Only rarely do all these fields need to be set!
 	 */
 	public function saveSettings() {
-		global $wgAuth;
 		$this->load();
 		if ( wfReadOnly() ) { return; }
 		if ( 0 == $this->mId ) { return; }
 
 		$this->mTouched = self::newTouchedTimestamp();
-		if ( !$wgAuth->allowSetLocalPassword() ) {
-			$this->mPassword = '';
-		}
 
 		wfRunHooks( 'BeforeUserSaveSettings', array( $this ) );
 
 		$dbw = wfGetDB( DB_MASTER );
 		$dbw->update( 'user',
 			array( /* SET */
-				'user_name' => $this->mName,
-				'user_password' => $this->mPassword,
-				'user_newpassword' => $this->mNewpassword,
-				'user_newpass_time' => $dbw->timestampOrNull( $this->mNewpassTime ),
-				'user_real_name' => $this->mRealName,
-				'user_email' => $this->mEmail,
-				'user_email_authenticated' => $dbw->timestampOrNull( $this->mEmailAuthenticated ),
-				'user_touched' => $dbw->timestamp( $this->mTouched ),
-				'user_token' => strval( $this->mToken ),
-				'user_email_token' => $this->mEmailToken,
-				'user_email_token_expires' => $dbw->timestampOrNull( $this->mEmailTokenExpires ),
+				   'user_name' => $this->mName,
+				   'user_newpassword' => $this->mNewpassword,
+				   'user_newpass_time' => $dbw->timestampOrNull( $this->mNewpassTime ),
+				   'user_real_name' => $this->mRealName,
+				   'user_email' => $this->mEmail,
+				   'user_email_authenticated' => $dbw->timestampOrNull( $this->mEmailAuthenticated ),
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "user_email_authenticated" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)